src/Symfony/Component/HttpFoundation/Session/Storage/Handler/RedisSessionHandler.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Session\Storage\Handler;
+
+/**
+ * Redis based session storage handler based on the Redis class
+ * provided by the PHP redis extension.
+ *
+ * Direct port of MemcachedSessionHandler to Redis.
+ *
+ * @see http://php.net/redis
+ *
+ * @author Drak <drak@zikula.org>
+ * @author Dalibor Karlović <dkarlovi@gmail.com>
+ */
+class RedisSessionHandler extends AbstractSessionHandler
+{
+    /**
+     * @var \Redis
+     */
+    private $redis;
+
+    /**
+     * @var int Time to live in seconds
+     */
+    private $ttl;
+
+    /**
+     * @var string Key prefix for shared environments
+     */
+    private $prefix;
+
+    /**
+     * List of available options:
+     *  * prefix: The prefix to use for the Redis keys in order to avoid collision
+     *  * expiretime: The time to live in seconds.
+     *
+     * @param \Redis $redis   A \Redis instance
+     * @param array  $options An associative array of Redis options
+     *
+     * @throws \InvalidArgumentException When unsupported options are passed
+     */
+    public function __construct(\Redis $redis, array $options = null)
+    {
+        $this->redis = $redis;
+
+        if ($diff = array_diff(array_keys($options), array('prefix', 'expiretime'))) {
+            throw new \InvalidArgumentException(sprintf(
+                'The following options are not supported "%s"', implode(', ', $diff)
+            ));
+        }
+
+        $this->ttl = isset($options['expiretime']) ? (int) $options['expiretime'] : 86400;
+        $this->prefix = isset($options['prefix']) ? $options['prefix'] : 'sf2s';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doRead($sessionId)
+    {
+        return $this->redis->get($this->prefix.$sessionId) ?: '';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doWrite($sessionId, $data)
+    {
+        return $this->redis->set($this->prefix.$sessionId, $data, $this->ttl);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDestroy($sessionId)
+    {
+        // number of deleted items
+        $count = $this->redis->del($this->prefix.$sessionId);
+
+        return 1 === $count;
```

Comment:
This is sensitive to race conditions. I think this should always return true instead.

Reply:
Will change to `return true`, but FMI, how's this prone to race conditions?

Reply:
if two concurrent calls to del() are made, one of them will fail because the item will be already deleted